datalake/src/main/java/com/sequenceiq/datalake/service/sdx/dr/SdxDatabaseDrService.java

```diff
@@ -0,0 +1,309 @@
+            throws JsonProcessingException {
+        StackV4Response stackV4Response = stackV4Endpoint.get(0L, sdxCluster.getClusterName(), Collections.emptySet());
+        LOGGER.info("Response from cloudbreak: {}", JsonUtil.writeValueAsString(stackV4Response));
+        ClusterV4Response cluster = stackV4Response.getCluster();
+        if (isStackOrClusterDrStatusComplete(stackV4Response.getStatus()) && isStackOrClusterDrStatusComplete(cluster.getStatus())) {
+            return sdxDrSucceeded(stackV4Response);
+        } else if (isStackOrClusterStatusFailed(stackV4Response.getStatus())) {
+            LOGGER.info("{} failed for Stack {} with status {}", pollingMessage, stackV4Response.getName(), stackV4Response.getStatus());
+            return sdxDrFailed(sdxCluster, stackV4Response.getStatusReason(), pollingMessage);
+        } else if (isStackOrClusterStatusFailed(stackV4Response.getCluster().getStatus())) {
+            LOGGER.info("{} failed for Cluster {} status {}", pollingMessage, stackV4Response.getCluster().getName(),
+                    stackV4Response.getCluster().getStatus());
+            return sdxDrFailed(sdxCluster, stackV4Response.getCluster().getStatusReason(), pollingMessage);
+        } else if (FINISHED.equals(flowState)) {
+            LOGGER.info("Flow finished, but Backup/Restore is not complete: {}", sdxCluster.getClusterName());
+            return sdxDrFailed(sdxCluster, "stack is in improper state", pollingMessage);
+        } else {
+            LOGGER.info("Flow is unknown state");
+            return sdxDrFailed(sdxCluster, "Flow is unknown state", pollingMessage);
+        }
+    }
+
+    /**
+     * Checks the status if the backup/restore operation failed.
+     * @param status It could be stack/cluster status
+     * @return true if status is failed, false otherwise.
+     */
+    private boolean isStackOrClusterStatusFailed(Status status) {
+        return (Status.BACKUP_FAILED.equals(status) ||
+                Status.RESTORE_FAILED.equals(status)) ? true : false;
+    }
+
+    private AttemptResult<StackV4Response> sdxDrFailed(SdxCluster sdxCluster, String statusReason, String pollingMessage) {
+        LOGGER.info("{} failed, statusReason: {}", pollingMessage, statusReason);
+        return AttemptResults.breakFor("SDX " + pollingMessage + " failed '" + sdxCluster.getClusterName() + "', " + statusReason);
+    }
+
+    private AttemptResult<StackV4Response> sdxDrSucceeded(StackV4Response stackV4Response) {
+        LOGGER.info("Database DR operation does SDX cluster {} is successful", stackV4Response.getCluster().getName());
+        return AttemptResults.finishWith(stackV4Response);
+    }
+
+    /**
+     * Checks the status if the backup/restore operation is successful.
+     * @param status It could be stack/cluster status
+     * @return true if status is successful, false otherwise.
+     */
```

Comment:
drop the comment, method name is self explaining

Reply:
will do